single-node-refactor/src/Solvers/SGH_solver_rz/src/position_rz.cpp

```diff
@@ -59,6 +59,8 @@ void SGHRZ::update_position_rz(double rk_alpha,
         for (int dim = 0; dim < num_dims; dim++) {
             double half_vel = (node_vel(1, node_gid, dim) + node_vel(0, node_gid, dim)) * 0.5;
             node_coords(1, node_gid, dim) = node_coords(0, node_gid, dim) + rk_alpha * dt * half_vel;
+
+            node_coords(1, node_gid, dim) = fmax(0.0, node_coords(1, node_gid, dim));
```

Comment:
This limits our meshes to have a theta value bounded by [0,90], but our mesher allows for the starting and ending angle to be outside this range.  There are two options as I see it:
1. Limit our mesher and add checks on read in meshes. 
2. Save an initial direction vector as ints [e.g. (1,1),(1,-1),(-1,-1),(-1,1)]  to encode the quadrant the node is in initially, and check against that to make sure it has not flipped.

Reply:
No, this is not theta.  There is no theta in the solver.  This max is on radius, and the radius in RZ-coordinates must be great than zero.  This max will not impact any mesh option in RZ-coordinates

Reply:
Are the node coordinates all radii, or are they stored as (x,y)?  If they are stored as (x,y), then this max is on (x,y).  If we dont allow x or y to be negative, then nodes below the (x,y) = (0,0) plane will all be forced onto the plane.

Reply:
the coordinates are: (z,r).  Z can be negative, but r cannot ever be negative

Reply:
Yes, any node that has r<0 value, then it must be forced to the axis of symmetry.  Otherwise the solver with give unphysical results at best (volume and mass are negative), more likely, a negative radius will give nans.

Reply:
I see that there is a bug in the coding, the max should only operate on the radius, dim=1.